specs/sharding/beacon-chain.md

```diff
@@ -652,27 +739,35 @@ The goal is to ensure that a proof can only be constructed if `deg(B) < l` (ther
 
 ```python
 def process_shard_proposer_slashing(state: BeaconState, proposer_slashing: ShardProposerSlashing) -> None:
-    reference_1 = proposer_slashing.signed_reference_1.message
-    reference_2 = proposer_slashing.signed_reference_2.message
-
-    # Verify header slots match
-    assert reference_1.slot == reference_2.slot
-    # Verify header shards match
-    assert reference_1.shard == reference_2.shard
-    # Verify header proposer indices match
-    assert reference_1.proposer_index == reference_2.proposer_index
-    # Verify the headers are different (i.e. different body)
-    assert reference_1 != reference_2
+    # Verify the headers are different
+    assert proposer_slashing.body_root_1 != proposer_slashing.body_root_2
```

Comment:
Body roots can be equal but `builder_index_1 != builder_index_2`. I would suggest to 
```py
     assert reference_1 != reference_2
```
(if this is correct comparison in Py)